py/desispec/pipeline/plan.py

```diff
@@ -398,7 +398,8 @@ def graph_night(rawdir, rawnight):
             node['out'] = []
             grph[name] = node
 
-    for name, nd in grph.items():
+    current_items = list(grph.items())
```

Comment:
Why do you need to convert to a list first?  dict.items() is still an iterable in Python 3.